src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final String EXCAVATOR_SIMPLE_CLASS_NAME = "Excavator";
+    private static final String TRUCK_SIMPLE_CLASS_NAME = "Truck";
+    private static final String BULLDOZER_SIMPLE_CLASS_NAME = "Bulldozer";
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+
+        switch (type.getSimpleName()) {
+            case EXCAVATOR_SIMPLE_CLASS_NAME:
+                machines = new ExcavatorProducer().get();
+                break;
+            case TRUCK_SIMPLE_CLASS_NAME:
+                machines = new TruckProducer().get();
+                break;
+            case BULLDOZER_SIMPLE_CLASS_NAME:
+                machines = new BulldozerProducer().get();
+                break;
+            default:
+                machines = new ArrayList<>();
+        }
+        return new ArrayList<>(machines);
```

Comment:
This code must be a good example of polymorphism.
Let's create an abstract variable of a type MachineProducer and initialize it with an applicable producer implementation. Then just call the method get one time.

------
(optional) Also, you can try to use a hashMap instead of if-blocks. It is a really good practice.